File: srv/models/news.py
from sqlmodel import Field, SQLModel, Relationship

class News(SQLModel, table=True):
    id: int | None = Field(default=None, primary_key=True)
    author: str | None = Field(index=True)
    title: str | None = Field(index=True)
    description: str | None = Field()
    url: str | None = Field()
    publishedAt: str | None = Field()
    content: str | None = Field()

File: frontend/app/model/news.ts
export class NewsData {
    id: number;
    author: string;
    title: string;
    description: string;
    url: string;
    publishedAt: string;
    content: string;
    constructor(id: number, author: string, title: string, description: string, url: string, publishedAt: string, content: string) {
        this.id = id;
        this.author = author;
        this.title = title;
        this.description = description;
        this.url = url;
        this.publishedAt = publishedAt;
        this.content = content;
    }
}

export default NewsData;

